src/document/types.ts

```diff
@@ -67,13 +67,13 @@ export type ImmutableReducer<State, A extends Action, LocalState> = (
     state: Draft<Document<State, A, LocalState>>,
     action: A | BaseAction,
     dispatch?: SignalDispatch,
-) => Document<State, A, LocalState> | void;
+) => Document<State, A, LocalState> | undefined;
```

Comment:
@acaldas I made these changes because of this rule https://typescript-eslint.io/rules/no-invalid-void-type/

let me know if the change doesn't make sense and I'll revert